packages/magento-product/components/StickyAddToCart/StickyAddToCart.tsx

```diff
@@ -0,0 +1,163 @@
+import { Image } from '@graphcommerce/image'
+import { LayoutHeader } from '@graphcommerce/next-ui'
+import { Box, Typography } from '@mui/material'
+import { useEffect, useState } from 'react'
+import { ProductListItemFragment } from '../../Api/ProductListItem.gql'
+import { AddProductsToCartButton, AddProductsToCartError } from '../AddProductsToCart'
+import { UseAddProductsToCartActionFragment } from '../AddProductsToCart/UseAddProductsToCartAction.gql'
+import { ProductPageAddToCartQuantityRow } from '../ProductPage/ProductPageAddToCartRow'
+import { ProductPageGalleryFragment } from '../ProductPageGallery/ProductPageGallery.gql'
+import { ProductPageName } from '../ProductPageName'
+import { ProductPagePrice } from '../ProductPagePrice'
+
+type StickyAddToCartProps = {
+  product: UseAddProductsToCartActionFragment & ProductPageGalleryFragment & ProductListItemFragment
+  cartButtonRef?: React.RefObject<HTMLButtonElement>
+}
+
+export function StickyAddToCart(props: StickyAddToCartProps) {
+  const { product, cartButtonRef } = props
+  const mainImage = product?.media_gallery?.[0]
+  const [isButtonInViewport, setIsButtonInViewport] = useState(false)
+
+  useEffect(() => {
+    if (!cartButtonRef || !cartButtonRef.current) return
+
+    const options = {
+      root: null,
+      rootMargin: '0px',
+      threshold: 0,
+    }
+
+    const observer = new IntersectionObserver((entries) => {
+      entries.forEach((entry) => {
+        if (entry.isIntersecting) {
+          setIsButtonInViewport(true)
+        } else {
+          setIsButtonInViewport(false)
+        }
+      })
+    }, options)
+
+    observer.observe(cartButtonRef.current)
+
+    // eslint-disable-next-line consistent-return
+    return () => {
+      observer.disconnect()
+    }
+  }, [cartButtonRef, isButtonInViewport])
+
+  if (!product?.sku || !product.url_key) return null
+
+  return (
+    <LayoutHeader
+      hideBackButton
+      forceScrolled={isButtonInViewport}
+      size='responsive'
+      sx={(theme) => ({
+        width: { md: '60vw', lg: '65vw' },
+        inset: 0,
+        margin: 'auto',
+        transform: {
+          xs: 'none',
+          md: `translateY(calc(${theme.appShell.appBarHeightMd}  * .28))`,
```

Comment:
Where does this calculation come from? Why do you multiply the `appBarHeight` by .28?